src/main/java/com/iota/iri/service/API.java

```diff
@@ -912,4 +817,218 @@ public void shutDown() {
             server.stop();
         }
     }
+
+    // TransactionViewModel, ApproveeViewModel, AddressViewModel, BundleViewModel, TagViewModel Interface
+    private TransactionViewModel transactionViewModelFromHash(final Tangle tangle, final Hash hash) {
+        try {
+            return TransactionViewModel.fromHash(tangle, hash);
+        } catch (Exception e) {
+            throw new IllegalStateException(e.getMessage());
+        }
+    }
+
+    private ApproveeViewModel approveeViewModelGetApprovers(final TransactionViewModel transactionViewModel, final Tangle tangle) {
+        try {
+            return transactionViewModel.getApprovers(tangle);
+        } catch (Exception e) {
+            throw new IllegalStateException(e.getMessage());
+        }
+    }
+
+    private boolean transactionViewModelExist(final Tangle tangle, final Hash hash) {
+        try {
+            return TransactionViewModel.exists(tangle, hash);
+        } catch (Exception e) {
+            throw new IllegalStateException(e.getMessage());
+        }
+    }
+
+    private AddressViewModel addressViewModelLoad(final Tangle tangle, final Indexable indexable) {
```

Comment:
still not sure what you're trying to achieve with these local wrappers, this only creates a bunch of unnecessary symbols

Reply:
is it about converting the exceptions (whatever they might be) into `IllegalStateException`s?
smells funky.

Reply:
This funky approach is nice here. Each request needs a response, thats for sure, with 2xx, 3xx, 4xx, 5xx status code. We can just rise an exception, wherever in the code (IllegalArgumentException for errors in the input, IllegalStateExceptions for internal errors) and handle all exceptions in just one place. The idea is: We don't need to propagate all compile time exceptions in every single function because we know where to handle all the errors and what to do with them (log the error and send the response). This wraper could represents a thin layer between the API and the other components. it could be in another file.